cmd/ci-operator-checkconfig/main.go

```diff
@@ -68,11 +74,20 @@ func (o *options) validate() (ret []error) {
 	inputCh := make(chan workItem)
 	produce := func() error {
 		defer close(inputCh)
-		if err := o.OperateOnCIOperatorConfigDir(o.ConfigDir, func(configuration *api.ReleaseBuildConfiguration, repoInfo *config.Info) error {
-			inputCh <- workItem{configuration, repoInfo}
-			return nil
-		}); err != nil {
-			return fmt.Errorf("error reading configuration files: %w", err)
+		for _, v := range o.ciOPConfigAgent.GetAll() {
+			for _, configs := range v {
+				for _, c := range configs {
+					c := c
+					configFilePath := filepath.Join(o.ConfigDir, c.Metadata.RelativePath())
+					configSpecDir := filepath.Dir(configFilePath)
+					inputCh <- workItem{configuration: &c, repoInfo: &config.Info{
+						Metadata: c.Metadata,
+						Filename: configFilePath,
+						OrgPath:  filepath.Dir(configSpecDir),
+						RepoPath: configSpecDir,
+					}}
```

Comment:
This is not part of this pull request, but since we now have all this extra code…  `repoInfo` is only used by `validateTags`, and only the `Org`/`Repo`/`Branch`/`Variant` members are used there.  We can replace it by `api.Metadata` and avoid these path calculations.